MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult.swift

```diff
@@ -75,6 +75,7 @@ import Foundation
     ///   - forceRefresh: Ignore any existing access token in the cache and force MSAL to get a new access token from the service.
     ///   - correlationId: Optional. UUID to correlate this request with the server for debugging.
     ///   - delegate: Delegate that receives callbacks for the Get Access Token flow.
+    @available(*, deprecated, message: "Use the 'getAccessToken' method that also receives a list of scopes to be included in the token")
```

Comment:
For a user that simply wants to do `result.getAccessToken(delegate: self)` this is marked as deprecated because Swift doesn't have a way to know which `getAccessToken` method the user is calling. The only way to silence this warning is by passing scopes.

![Screenshot 2024-05-13 at 14 15 01](https://github.com/AzureAD/microsoft-authentication-library-for-objc/assets/109726904/8801b46e-89e0-480e-ae4f-9eda3d1c2d44)

IMO this worsens the Swift developer experience, because it gives the impression that they are using a deprecated method (when they're not) in those use cases that don't require more scopes.

However if we don't add a second method, this could become a breaking change for objective-c users. Ideally we would have the following:

1. In the same Swift method, we add the `scopes` parameter with a default value. This method will be see from both Swift and Objc.
2. We create an obj-only "old" method without the scopes and we deprecated it, so the existing Objc users will see be able to continue using it and will see the deprecation warning. I'm not sure if this is possible or not tbh. Can you check it?

Reply:
Hey Diego, this method will not be deprecated anymore :)
Marcos is going to update the PR.

Reply:
Nice. But what happens with the existing objc users that may have used it @nilo-ms ?

Reply:
Sorry my bad, the way it is now won't affect existing users 👍